src/NUnitEngine/nunit.engine.core/Drivers/NUnitFrameworkApi2009.cs

```diff
@@ -0,0 +1,157 @@
+            Type? _frameworkControllerType;
+
+            public NUnitFrameworkApi2009(AppDomain testDomain, string driverId, AssemblyName nunitRef)
+            {
+                Guard.ArgumentNotNull(testDomain, nameof(testDomain));
+                Guard.ArgumentNotNull(driverId, nameof(driverId));
+                Guard.ArgumentNotNull(nunitRef, nameof(nunitRef));
+
+                _testDomain = testDomain;
+                _driverId = driverId;
+                _nunitRef = nunitRef;
+            }
+
+            public string Load(string testAssemblyPath, IDictionary<string, object> settings)
+            {
+                Guard.ArgumentValid(File.Exists(testAssemblyPath), "Framework driver called with a file name that doesn't exist.", "testAssemblyPath");
+
+                log.Info($"Loading {testAssemblyPath} - see separate log file");
+
+                _testAssemblyPath = testAssemblyPath;
+
+                // Normally, the caller should check for an invalid requested runtime, but we make sure here
+                var requestedRuntime = settings.ContainsKey(EnginePackageSettings.RequestedRuntimeFramework)
+                    ? settings[EnginePackageSettings.RequestedRuntimeFramework] : null;
+
+                var idPrefix = string.IsNullOrEmpty(_driverId) ? "" : _driverId + "-";
+
+                try
+                {
+                    _frameworkController = CreateObject(CONTROLLER_TYPE, _testAssemblyPath, idPrefix, settings);
+                }
+                catch (BadImageFormatException ex) when (requestedRuntime != null)
+                {
+                    throw new NUnitEngineException($"Requested runtime {requestedRuntime} is not suitable for use with test assembly {_testAssemblyPath}", ex);
+                }
+                catch (SerializationException ex)
+                {
+                    throw new NUnitEngineException("The NUnit 3 driver cannot support this test assembly. Use a platform specific runner.", ex);
+                }
+
+                _frameworkControllerType = _frameworkController.GetType();
+                log.Debug($"Created FrameworkController {_frameworkControllerType.Name}");
+
+                return ExecuteAction(LOAD_ACTION);
+            }
+
+            public int CountTestCases(string filter)
+            {
+                CheckLoadWasCalled();
+                return int.Parse(ExecuteAction(COUNT_ACTION, filter));
+            }
+
+            public string Run(ITestEventListener? listener, string filter)
+            {
+                CheckLoadWasCalled();
+                log.Info("Running {0} - see separate log file", Path.GetFileName(_testAssemblyPath.ShouldNotBeNull()));
+                return ExecuteAction(RUN_ACTION, listener, filter);
+            }
+
+            public void RunAsync(Action<string> callback, string filter) => throw new NotImplementedException();
+
+            public void StopRun(bool force) => ExecuteAction(STOP_RUN_ACTION, force);
```

Comment:
No check on `CheckLoadWasCalled`. Same comment as for 2018 driver. Move that check into the `ExecuteAction` method.